### PR TITLE
Upgrade documentation to v13.2.5

### PR DESCRIPTION
**Pull request by JannikGM**
**This was barely tested; please try it out**

The change in related.js was probably made necessary by https://github.com/documentationjs/documentation/commit/48bb7a283e782b2398e7b4061b5edbcf3367bb87#diff-5cd6e4a33443db542ccef69418146704eed1596b7b14be9684d67471bb836451R234

### CODE REVIEW

**Comment by astridx** [comment]
I just tested the documentation-13.x branch. I had no problems with the current main branch of maplibre-gl-js.

```

$ git fetch origin  pull/86/head:documentation-13.x
remote: Enumerating objects: 8, done.
remote: Counting objects: 100% (8/8), done.
remote: Total 8 (delta 7), reused 8 (delta 7), pack-reused 0
Unpacking objects: 100% (8/8), 19.89 KiB | 278.00 KiB/s, done.
From https://github.com/maplibre/maplibre-gl-js-docs
 * [new ref]         refs/pull/86/head -> documentation-13.x
astrid@ubuntu:~/git/maplibre/maplibre-gl-js-docs$ git checkout documentation-13.x
Switched to branch 'documentation-13.x'


$ npm ci

added 3173 packages, and audited 3174 packages in 3m

61 vulnerabilities (5 low, 22 moderate, 31 high, 3 critical)

To address issues that do not require attention, run:
  npm audit fix

To address all issues possible (including breaking changes), run:
  npm audit fix --force

Some issues need review, and may require choosing
a different dependency.

Run `npm audit` for details.



$ git submodule update --init


$ nvm use && npm start
Found '/home/astrid/git/maplibre/maplibre-gl-js-docs/.nvmrc' with version <12>
Now using node v12.18.3 (npm v7.6.1)

> prestart
> npm run build-docs


> build-docs
> run-s install-gl-js-deps build-api build-images


> install-gl-js-deps
> cd maplibre-gl-js/ && npm install --ignore-scripts --no-package-lock --no-audit --loglevel=error


up to date in 5s

122 packages are looking for funding
  run `npm fund` for details

> build-api
> documentation build --github --format json --sort-order alpha --config ./docs/documentation.yml --output docs/components/api.json maplibre-gl-js/src/index.js

Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db

Why you should do it regularly:
https://github.com/browserslist/browserslist#browsers-data-updating

> build-images
> mkdir -p docs/img/dist && node docs/bin/build-image-config.js && node docs/bin/appropriate-images.js --all

Saved 3d-buildings-500.png
...
Saved zoomto-linestring-800.webp
Generated 508 optimized images

> start
> NODE_OPTIONS="--max_old_space_size=4096" DEPLOY_ENV=local batfish start

[10:20:20 Batfish] Starting the development server.
[10:20:20 Batfish] Wait ...
Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db

Why you should do it regularly:
https://github.com/browserslist/browserslist#browsers-data-updating
[10:22:51 Batfish] Go!
[10:22:51 Batfish] Development server ready.
  > Access your site at http://localhost:8080/maplibre-gl-js-docs/
  > Available externally at http://192.168.209.132:8080/maplibre-gl-js-docs/
  > Compiled files are in _site
[10:22:51 Batfish] Webpack compiled in 2m 27.7s
[10:22:56 Batfish] Webpack compiled in 3.2s
 

```

**Comment by astridx** [comment]
I tested the documentation-13.x branch at the same time as the typescipt branch of maplibre-gl-js. Here I got errors.


```
/maplibre-gl-js-docs$ cd maplibre-gl-js/

/maplibre/maplibre-gl-js-docs/maplibre-gl-js$ git checkout typescript 
Previous HEAD position was efad8e0c2 fix breaking changes to css classnames (#203)
Switched to branch 'typescript'
Your branch is up-to-date with 'origin/typescript'.

git/maplibre/maplibre-gl-js-docs/maplibre-gl-js$ git pull origin typescript 
From https://github.com/maplibre/maplibre-gl-js
 * branch                typescript -> FETCH_HEAD
Already up-to-date.


/maplibre/maplibre-gl-js-docs/maplibre-gl-js$ cd ..

/maplibre/maplibre-gl-js-docs$ nvm use && npm start
Found '/maplibre/maplibre-gl-js-docs/.nvmrc' with version <12>
Now using node v12.18.3 (npm v7.6.1)

> prestart
> npm run build-docs


> build-docs
> run-s install-gl-js-deps build-api build-images


> install-gl-js-deps
> cd maplibre-gl-js/ && npm install --ignore-scripts --no-package-lock --no-audit --loglevel=error


added 638 packages, removed 125 packages, and changed 72 packages in 2m

154 packages are looking for funding
  run `npm fund` for details

> build-api
> documentation build --github --format json --sort-order alpha --config ./docs/documentation.yml --output docs/components/api.json maplibre-gl-js/src/index.js

Table of contents defined sorting of Map but no documentation with that namepath was found
Table of contents defined sorting of accessToken but no documentation with that namepath was found
Table of contents defined sorting of baseApiUrl but no documentation with that namepath was found
Table of contents defined sorting of workerCount but no documentation with that namepath was found
Table of contents defined sorting of maxParallelImageRequests but no documentation with that namepath was found
Table of contents defined sorting of supported but no documentation with that namepath was found
Table of contents defined sorting of version but no documentation with that namepath was found
Table of contents defined sorting of setRTLTextPlugin but no documentation with that namepath was found
Table of contents defined sorting of getRTLTextPluginStatus but no documentation with that namepath was found
Table of contents defined sorting of clearStorage but no documentation with that namepath was found
Table of contents defined sorting of AnimationOptions but no documentation with that namepath was found
Table of contents defined sorting of CameraOptions but no documentation with that namepath was found
Table of contents defined sorting of PaddingOptions but no documentation with that namepath was found
Table of contents defined sorting of RequestParameters but no documentation with that namepath was found
Table of contents defined sorting of StyleImageInterface but no documentation with that namepath was found
Table of contents defined sorting of CustomLayerInterface but no documentation with that namepath was found
Table of contents defined sorting of prewarm but no documentation with that namepath was found
Table of contents defined sorting of clearPrewarmedResources but no documentation with that namepath was found
Table of contents defined sorting of Marker but no documentation with that namepath was found
Table of contents defined sorting of Popup but no documentation with that namepath was found
Table of contents defined sorting of IControl but no documentation with that namepath was found
Table of contents defined sorting of NavigationControl but no documentation with that namepath was found
Table of contents defined sorting of GeolocateControl but no documentation with that namepath was found
Table of contents defined sorting of AttributionControl but no documentation with that namepath was found
Table of contents defined sorting of ScaleControl but no documentation with that namepath was found
Table of contents defined sorting of FullscreenControl but no documentation with that namepath was found
Table of contents defined sorting of LngLat but no documentation with that namepath was found
Table of contents defined sorting of LngLatLike but no documentation with that namepath was found
Table of contents defined sorting of LngLatBounds but no documentation with that namepath was found
Table of contents defined sorting of LngLatBoundsLike but no documentation with that namepath was found
Table of contents defined sorting of Point but no documentation with that namepath was found
Table of contents defined sorting of PointLike but no documentation with that namepath was found
Table of contents defined sorting of MercatorCoordinate but no documentation with that namepath was found
Table of contents defined sorting of EdgeInsets but no documentation with that namepath was found
Table of contents defined sorting of BoxZoomHandler but no documentation with that namepath was found
Table of contents defined sorting of ScrollZoomHandler but no documentation with that namepath was found
Table of contents defined sorting of DragPanHandler but no documentation with that namepath was found
Table of contents defined sorting of DragRotateHandler but no documentation with that namepath was found
Table of contents defined sorting of KeyboardHandler but no documentation with that namepath was found
Table of contents defined sorting of DoubleClickZoomHandler but no documentation with that namepath was found
Table of contents defined sorting of TouchZoomRotateHandler but no documentation with that namepath was found
Table of contents defined sorting of TouchPitchHandler but no documentation with that namepath was found
Table of contents defined sorting of GeoJSONSource but no documentation with that namepath was found
Table of contents defined sorting of VideoSource but no documentation with that namepath was found
Table of contents defined sorting of ImageSource but no documentation with that namepath was found
Table of contents defined sorting of CanvasSource but no documentation with that namepath was found
Table of contents defined sorting of CanvasSourceOptions but no documentation with that namepath was found
Table of contents defined sorting of Evented but no documentation with that namepath was found
Table of contents defined sorting of MapMouseEvent but no documentation with that namepath was found
Table of contents defined sorting of MapTouchEvent but no documentation with that namepath was found
Table of contents defined sorting of MapBoxZoomEvent but no documentation with that namepath was found
Table of contents defined sorting of MapDataEvent but no documentation with that namepath was found
Table of contents defined sorting of MapWheelEvent but no documentation with that namepath was found

> build-images
> mkdir -p docs/img/dist && node docs/bin/build-image-config.js && node docs/bin/appropriate-images.js --all

Saved 3d-buildings-500.png
...
Saved zoomto-linestring-800.webp
Generated 508 optimized images

> start
> NODE_OPTIONS="--max_old_space_size=4096" DEPLOY_ENV=local batfish start

/maplibre/maplibre-gl-js-docs/node_modules/@mapbox/batfish/bin/batfish.js:168
    throw error;
    ^

TypeError: Cannot read property 'params' of undefined
    at buildSubSubNav (/git/maplibre/maplibre-gl-js-docs/docs/data/api-navigation.js:96:14)
    at Object.<anonymous> (/maplibre/maplibre-gl-js-docs/docs/data/api-navigation.js:38:17)
    at Module._compile (internal/modules/cjs/loader.js:1137:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1157:10)
    at Module.load (internal/modules/cjs/loader.js:985:32)
    at Function.Module._load (internal/modules/cjs/loader.js:878:14)
    at Module.require (internal/modules/cjs/loader.js:1025:19)
    at require (internal/modules/cjs/helpers.js:72:18)
    at Object.<anonymous> (/maplibre/maplibre-gl-js-docs/batfish.config.js:4:23)
    at Module._compile (internal/modules/cjs/loader.js:1137:30)


```


@JannikGM I think your main goal was to support this branch, right? Does that work for you? Am I doing something wrong?

**Comment by JannikGM** [comment]
> I tested the documentation-13.x branch with the typescipt branch of maplibre-gl-js, too. Here I got errors.
> [...]
> @JannikGM I think your main goal was to support this branch, right? Does that work for you? Am I doing something wrong?

No, this doesn't affect TypeScript (yet).

Initially, I thought we'd need documentation-13.x for TypeScript support, because [that's when they added a TS example to the npm package README](https://github.com/documentationjs/documentation/commit/00b3478a2006287984b2703e7be246df33027b86).

However, after sending this PR I realized that TS support had existed much earlier in documentation.js, but with a different CLI.
Updating documentation.js is therefore (probably) independent of TS support.

It's still a good idea to keep our dependencies up-to-date and it will simplify how we can support TS.

I'll send a follow-up PR for the TS specific changes (with the documentation.js-13.x CLI). [See point 3 in my Slack message for TS specifics](https://osmus.slack.com/archives/C01G3D28DAB/p1629276611065200?thread_ts=1629109687.049300&cid=C01G3D28DAB) (ignore point 2, because it's incorrect).

**Comment by HarelM** [comment]
The changes looks straightforward, how can I help?

**Comment by wipfli** [comment]
https://github.com/maplibre/maplibre-gl-js/issues/92

**Comment by wipfli** [comment]
The fact that v2 is a pre-release might cause some problems. Let's see

**Comment by JannikGM** [comment]
> The changes looks straightforward, how can I help?

Review / ensure everything still works as intended (and find someone who's able and willing to merge it?).
Yes, tiny change, but upgrades a core dependency to a new major version.

> The fact that v2 is a pre-release might cause some problems. Let's see

How so? Will this auto-deploy for the latest maplibre-gl-js repo?

This PR still doesn't apply the TS changes (which I'll do in a follow-up if nobody else does it), so if CI automatically deploys the docs for the latest maplibre-gl-js repo, this would probably break now.

**Comment by wipfli** [comment]
Hi Jannik. I am reviewing this right now. Indeed, this pull request is not about typescript. So lets discuss this somewhere else.

**Comment by wipfli** [comment]
Don't worry about the CI. We deploy the version which is in the git submodule. To update this one, you have to manually run https://github.com/maplibre/maplibre-gl-js-docs/blob/main/.github/workflows/update-gl-js.yml or do it with git directly.

File: docs/components/api/related.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ export default class Related extends React.Component {
             <SectionWrapper title="Related" {...this.props}>
                 <ul>
                     {section.sees.map((see, i) => (
-                        <li key={i}>{toHtml(see, true)}</li>
+                        <li key={i}>{toHtml(see.description, true)}</li>
```
Comment (wipfli) [inline docs/components/api/related.js:13]: Why is this needed?

**Review by wipfli — APPROVED** [review]

**Comment by astridx** [comment]
@wipfli Did you see this in the first comment?

> The change in related.js was probably made necessary by [documentationjs/documentation@48bb7a2#diff-5cd6e4a33443db542ccef69418146704eed1596b7b14be9684d67471bb836451R234](https://github.com/documentationjs/documentation/commit/48bb7a283e782b2398e7b4061b5edbcf3367bb87#diff-5cd6e4a33443db542ccef69418146704eed1596b7b14be9684d67471bb836451R234)

File: docs/components/api/related.js (hunk 1 above)
Comment (astridx) [reply docs/components/api/related.js:13]: Did you see this in the first comment?

> The change in related.js was probably made necessary by [documentationjs/documentation@48bb7a2#diff-5cd6e4a33443db542ccef69418146704eed1596b7b14be9684d67471bb836451R234](https://github.com/documentationjs/documentation/commit/48bb7a283e782b2398e7b4061b5edbcf3367bb87#diff-5cd6e4a33443db542ccef69418146704eed1596b7b14be9684d67471bb836451R234)

**Comment by wipfli** [comment]
Thanks Astrid
